### ISSUE TITLE
Minor bug: implicit tuple syntax not available on certain functions

### ISSUE THREAD

**Issue by lgalabru** [blockstack/stacks-blockchain#1175]
In the context of BNS, I've noticed that the implicit syntax for constructing a tuple is not available here:
```
(nft-get-owner names (tuple (name name) (namespace namespace)))
```

**Comment by lgalabru** [comment]
Note: @jcnelson had the same issue with `(default-to ...`

@kantai do you think that we should we reconsider https://github.com/blockstack/blockstack-core/issues/986 ?

**Comment by kantai** [comment]
I think it could be doable for us, because we don't allow "anonymous" functions, there's no ambiguity in Clarity for the implicit tuple syntax when used in arbitrary locations. For example:

```
(let (coord ((x 10) (y 11) (z 12)))
   ... )
```

The _only_ possible interpretation of a list expression of list expressions is as an implicit tuple -- in the above case, the only legal interpretation is that `coord` is bound to a tuple with keys `x`,`y`,`z`.

If we pursue this, what I'd say is that we should call _implicit_ tuple specifications sugar, and use the sugar expander to turn them into _explicit_ tuple specifications. That way, the analysis passes and the vm execution don't have to do all of the checks for explicity vs. implicit tuples.

**Comment by njordhov** [comment]
> The only possible interpretation of a list expression of list expressions is as an implicit tuple

Such position dependent implicit syntax adds a cognitive burden on the programmer. Lisp dialects conventionally provide the quote special operator with the shortcut `'` to yield the un-evaluated form, which would be the tuple in Clarity:

      (let (coord '((x 10) (y 11) (z 12)))
         ... )

That's just one extra character to remove the ambiguity of the implicit syntax.

**Comment by lgalabru** [comment]
thank you for your feedback @njordhov, it sounds like a great idea!
we're already using the `'` char for some literals (principals, booleans), @kantai do you think that we can use the same character, or should we go with something else?

**Comment by njordhov** [comment]
I recognize that @kantai makes an interesting observation when noting that _the only possible interpretation of a list expression of list expressions is as an implicit tuple_ allowing using the implicit tuple syntax in arbitrary locations. This is important because it simplifies evaluation, avoiding having to process the arguments of every function as if it was a special form.

However, the construct overloads a convention in many Lisp dialects where it is used for calling a function returned from calling another function. This could be confusing for developers already experienced with Lisp languages. 

Although Clarity doesn't allow closures (presumably because they would open for recursion) it would be prudent to reserve the syntax for future uses, such as if clarity functions may return named top level functions to be called.

**Comment by njordhov** [comment]
> do you think that we can use the same character, or should we go with something else?

[Clojure](https://clojure.org) has demonstrated the benefits of extending s-expressions with carefully selected syntactic sugar. For such key/value tuples it provides a syntax much like what you find in e.g. Javascript, using curly braces. We could use a similar implicit notation for tuples in Clarity.

**Comment by kantai** [comment]
@njordhov — these are great ideas. I think eliminating ambiguity in the tuple syntax is a good idea. Using the single quote for this would be okay, though it may also be a little different from single quote usage in other lisps. I think the curly brace syntax may be the clearest approach.

**Comment by njordhov** [comment]
Resolving with PR #1320.

**Comment by njordhov** [comment]
Can be closed with #1234
